matflow/param_classes/load.py

```diff
@@ -871,6 +871,70 @@ def uniaxial_cyclic(
             out.extend(cycle_i)
         return [cls(**i)._remember_name_args(None, i) for i in out]
 
+    @classmethod
+    def from_npz_file(
+        cls,
+        npz_file_path,
+        idx,
+    ) -> list[Self]:
+        """
+        Read loading data from a numpy .npz file and format it appropriately.
+        Designed for running large arrays of simulations which each use loadcase of the
+        corresponding index.
+
+        Parameters
+        ----------
+        npz_file_path: str
+            Filepath to the loadcase.npz file.
+            File contains dict of the following keys:
+            num_incs: 1D numpy array
+                Array of the total increments of each loadcase.
+                (total number of increments the damask simulation should undergo)
+            inc_size: 2D numpy array
+                Array of the amount of strain each loadstep of each loadcase should undergo in the damask simulation
+            inc_size_final: 2D numpy array
+                Array of amount of strain of final loadsteps
+            u_sampled_split: 4D numpy array
+                Array of strain matricies (3x3) for each loadstep (1st,2nd D) should undergo for each loadcase (3nd D),
+                for each simulation (4th D). Sampled from the elements of an FE model.
```

Comment:
Isn't the load*case* the first dimension, the load step the second dimension, and the strain matrix the final dimensions?

Typo: `matrices`